jetty-io/src/main/java/org/eclipse/jetty/io/ChannelEndPoint.java

```diff
@@ -207,6 +207,8 @@ public void doClose()
             LOG.debug("doClose {}", this);
         try
         {
+            if (_key != null)
```

Comment:
`_key` is never `null`, so the check seems redundant, and is now also racy with `replaceKey()`.
I would remove the null check and perhaps `catch (Throwable)` below.

Reply:
`SocketChannelEndPointOpenCloseTest` fails without this check, but it passes a null `SelectionKey` to the `ChannelEndPoint` constructor.

I thought adding that null check was the safest thing to do, but you're right, I overlooked the `replaceKey()` mechanism which makes this null check odd.

I've changed the test so that it passes a non-null `SelectionKey` but I wonder if we shouldn't be stricter in the constructor about accepting null arguments when that should not be allowed.